cmd/argocd/commands/account.go

```diff
@@ -72,7 +74,7 @@ has appropriate RBAC permissions to change other accounts.
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
-			acdClient := argocdclient.NewClientOrDie(clientOpts)
+			acdClient := headless.NewClientOrDie(clientOpts, initialize.RetrieveContextIfChanged(c.Flag("context")))
```

Comment:
I think we can do this a little bit cleaner. Everywhere we repeat a call to `initialize.RetrieveContextIfChanged` when initializing the client. i.e.:

```golang
headless.NewClientOrDie(clientOpts, initialize.RetrieveContextIfChanged(c.Flag("context")))
```

Can we somehow improve/reduce this to:

```
headless.NewClientOrDie(clientOpts)
```

Can we update the clientOpts to have a `.context` field so we don't have to parse this everywhere?

Reply:
I was initially thinking of doing that but this specific context is only required when we need to create the local server and the CLientOptions struct is a more general client pkg data that might be used by other API clients and binaries so I felt it might be better to keep it specific to the cli only instead of modifying the original struct. But we can add it in there too.

Reply:
I see your point. But unfortunately, there's already a `clientOpts.Core` which is a useless flag to the generic argocd client package. So for better or worse, we already have CLI specific flag which made its way into in `clientOpts`.

If we want to be slightly cleaner about this, we could preserve the existing API client constructor. e.g.:

```golang
apiclient.NewClientOrDie(clientOpts)
```

And introduce a CLI specific constructor that accepts the `cobra.Command` as a second argument:

```golang
headless.NewClientOrDie(clientOpts, c)
```

The `headless.NewClientOrDie` would call:

```golang
initialize.RetrieveContextIfChanged(c.Flag("context"))
...
apiclient.NewClientOrDie(clientOpts)
```

Then we don't have to introduce more useless options to `clientOpts` which aren't really used by the API client package. And we could even remove the Core option from clientOpts. WDYT?

Reply:
That sounds good to me, also we'll shift the `headless.NewClientOrDie` to the CLI pkg so it will not technically change anything in the apiclient pkg.